### PR TITLE
Merge the multi-source work from @k0nserv / @hjanuschka

### PR DESCRIPTION
**Pull request by orta**
Alright, this takes the advice from https://github.com/danger/danger/pull/299#issuecomment-235624279

And is manually merging in the work necessary to allow others work on things like bitbucket or #382  👍

### CODE REVIEW

File: lib/danger/helpers/comments_helper.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
+module Danger
+  module Helpers
+    module CommentsHelper
+      def markdown_parser
+        @markdown_parser ||= Redcarpet::Markdown.new(Redcarpet::Render::HTML, no_intra_emphasis: true)
```
Comment (DanToml) [inline lib/danger/helpers/comments_helper.rb]: Might need to require red carpet here?

Comment (orta) [inline lib/danger/helpers/comments_helper.rb]: Think this is a good call

**Comment by k0nserv** [comment]
@orta You wanna pick 7f54b24 too it contains tests for `CommentsHelper` or I can do it and push it to this branch

**Comment by k0nserv** [comment]
It's intended to be squashed into 73754c2

**Comment by orta** [comment]
I'm not really a re-writing history kind of person

**Comment by orta** [comment]
👯

**Comment by k0nserv** [comment]
As in your are against it on a ideological basis?

**Comment by orta** [comment]
Nah, I just prefer to see the history as immutable - it's mostly laziness, so I've never bothered to learn

**Comment by k0nserv** [comment]
Okay do you want me to do it then? I prefer to see history as immutable after it reaches stables branches, but not in feature branches.

**Comment by orta** [comment]
Sure, you're welcome to it, 👍

**Comment by k0nserv** [comment]
All CI should pass now. If it does give me a shout when you are ready to merge and I'll rebase the branch and clean up the history a bit before :)

**Comment by k0nserv** [comment]
Okay also improved the test coverage of `CommentsHelper` so that at least all methods are tested. I think there's some extra coverage in the tests for the github request source as well.

**Comment by orta** [comment]
Awesome 💃

**Comment by orta** [comment]
This looks good to me - the windows CIs seem to have been queued for a day - so I'm OK with them not being green

**Comment by k0nserv** [comment]
Cool, I'll clean up the history and then merge

**Comment by k0nserv** [comment]
All except windows is green, merging this

**Pull request merged**